pkg/jx/cmd/add_app.go

```diff
@@ -224,83 +266,82 @@ func (o *AddAppOptions) Run() error {
 					log.Infof("Generated values.yaml:\n\n%v\n", util.ColorInfo(string(valuesYaml)))
 				}
 
-				// For each secret, we write a file into the chart
-				templatesDir := filepath.Join(dir, "templates")
-				err = os.MkdirAll(templatesDir, 0700)
-				if err != nil {
-					return err
-				}
-				for _, secret := range secrets {
-					file, err := ioutil.TempFile(templatesDir, fmt.Sprintf("%s-*.yaml", secret.Name))
-					defer func() {
-						err = file.Close()
-						if err != nil {
-							log.Warnf("Error closing %s because %v\n", file.Name(), err)
-						}
-					}()
-					if err != nil {
-						return err
-					}
-					bs, err := json.Marshal(secret)
-					if err != nil {
-						return err
-					}
-					ybs, err := yaml.JSONToYAML(bs)
-					if err != nil {
-						return err
-					}
-					_, err = file.Write(ybs)
-					if err != nil {
-						return err
-					}
-					if o.Verbose {
-						log.Infof("Added secret %s\n\n%v\n", secret.Name, util.ColorInfo(string(ybs)))
-					}
-					if err != nil {
-						return err
-					}
-				}
-				if !o.GitOps {
-					if len(o.ValueFiles) > 0 && schema != nil {
-						log.Warnf("values.yaml specified by --valuesFiles will be used despite presence of schema in app")
-					} else if schema != nil {
-						valuesFile, err := ioutil.TempFile("", fmt.Sprintf("%s-values.yaml", app))
-						defer func() {
-							err = valuesFile.Close()
+				// We write a secret template into the chart, append the values for the generated secrets to values.yaml
+				if len(secrets) > 0 {
+					if useVault {
+						for _, secret := range secrets {
+							path := strings.Join([]string{vaultBasepath, secret.Name}, "/")
+							err := vault.WriteMap(vaultClient, path, map[string]interface{}{
+								secret.Key: secret.Value,
+							})
 							if err != nil {
-								log.Warnf("Error closing %s because %v\n", valuesFile.Name(), err)
+								return err
 							}
-						}()
+						}
+					} else {
+						// For each secret, we write a file into the chart
+						templatesDir := filepath.Join(dir, "templates")
+						err = os.MkdirAll(templatesDir, 0700)
 						if err != nil {
 							return err
 						}
-						_, err = valuesFile.Write(valuesYaml)
+						fileName := filepath.Join(templatesDir, "app-generated-secret-template.yaml")
+						err := ioutil.WriteFile(fileName, []byte(secretTemplate), 0755)
 						if err != nil {
 							return err
 						}
-						o.ValueFiles = []string{
-							valuesFile.Name(),
+						allSecrets := map[string][]*surveyutils.GeneratedSecret{
+							appsGeneratedSecretKey: secrets,
+						}
+						ybs, err := json.Marshal(allSecrets)
+						if err != nil {
+							return err
 						}
+						valuesYaml = append(valuesYaml, ybs...)
```

Comment:
Is the `ybs` file which contains the secret going to be removed from the local file systems?

Reply:
No, thanks.